packages/compiler-dom/src/transforms/vOn.ts

```diff
@@ -1 +1,55 @@
-// TODO
+import {
+  transformOn as baseTransform,
+  DirectiveTransform,
+  createObjectProperty,
+  createCallExpression,
+  createObjectExpression,
+  createSimpleExpression
+} from '@vue/compiler-core'
+import { V_ON_MODIFIERS_GUARD } from '../runtimeHelpers'
+
+const EVENT_OPTION_MODIFIERS = { passive: true, once: true, capture: true }
+
+export const transformOn: DirectiveTransform = (dir, node, context) => {
+  const { modifiers } = dir
+  const baseResult = baseTransform(dir, node, context)
+  if (!modifiers.length) return baseResult
+  const { key, value } = baseResult.props[0]
+  const properties = [
+    createObjectProperty(
+      'handler',
+      createCallExpression(context.helper(V_ON_MODIFIERS_GUARD), [
+        value,
+        JSON.stringify(modifiers.filter(m => !(m in EVENT_OPTION_MODIFIERS)))
+      ])
+    ),
+    createObjectProperty(
+      'persistent',
+      createSimpleExpression(true as any, true)
```

Comment:
How do I create a primitive expression without breaking everything😂
- Let `SimpleExpressionNode['content']` support `boolean` (maybe also `number`)?
- Add `PrimitiveExpressionNode`+`NodeTypes.PRIMITIVE_EXPRESSION` and implement its AST utils / code gen?

Reply:
`createSimpleExpression(`true`, false)`